ui/components/multichain/multi-srp/import-srp/import-srp.test.tsx

```diff
@@ -0,0 +1,178 @@
+    clipboardData: {
+      getData: () => srp,
+    },
+  });
+
+  fireEvent(firstSRPWordInput, pasteEvent);
+};
+
+describe('ImportSRP', () => {
+  const store = configureMockStore([thunk])(mockState);
+
+  beforeEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it('enables the "Import wallet" button when a valid secret recovery phrase is entered', async () => {
+    const render = renderWithProvider(
+      <ImportSRP onActionComplete={jest.fn()} />,
+      store,
+    );
+    const { getByText } = render;
+
+    expect(getByText('Import wallet')).not.toBeEnabled();
+    await pasteSRPIntoFirstInput(render, VALID_SECRET_RECOVERY_PHRASE);
+
+    await waitFor(() => {
+      expect(getByText('Import wallet')).toBeEnabled();
+    });
+  });
+
+  it('does not enable the "Import wallet" button when the secret recovery phrase is empty', async () => {
+    const render = renderWithProvider(
+      <ImportSRP onActionComplete={jest.fn()} />,
+      store,
+    );
+    const { getByText } = render;
+
+    expect(getByText('Import wallet')).not.toBeEnabled();
+    await pasteSRPIntoFirstInput(render, '');
+    expect(getByText('Import wallet')).not.toBeEnabled();
+  });
+
+  it('calls addNewMnemonicToVault and showAlert on successful import', async () => {
+    const onActionComplete = jest.fn();
+    const render = renderWithProvider(
+      <ImportSRP onActionComplete={onActionComplete} />,
+      store,
+    );
+    const { getByText } = render;
+
+    expect(getByText('Import wallet')).not.toBeEnabled();
+    await pasteSRPIntoFirstInput(render, VALID_SECRET_RECOVERY_PHRASE);
+
+    fireEvent.click(getByText('Import wallet'));
+
+    await waitFor(() => {
+      expect(actions.importMnemonicToVault).toHaveBeenCalledWith(
+        VALID_SECRET_RECOVERY_PHRASE,
+      );
+      const dispatchedActions = store.getActions();
+      expect(dispatchedActions[0]).toStrictEqual({
+        type: 'SET_SHOW_NEW_SRP_ADDED_TOAST',
+        payload: true,
+      });
+      expect(onActionComplete).toHaveBeenCalledWith(true);
+    });
+  });
+
+  it('displays an error if one of the words in the srp is incorrect', async () => {
+    const onActionComplete = jest.fn();
+    const render = renderWithProvider(
+      <ImportSRP onActionComplete={onActionComplete} />,
+      store,
+    );
+    const { getByText, getByTestId } = render;
+
+    const invalidSRP = VALID_SECRET_RECOVERY_PHRASE.replace('input', 'inptu');
+    expect(getByText('Import wallet')).not.toBeEnabled();
+    await pasteSRPIntoFirstInput(render, invalidSRP);
+
+    // This is safe because the input is always present in the word div.
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const invalidWord = getByTestId('import-srp__srp-word-0').querySelector(
+      'input',
+    )!;
+
+    const importButton = getByText('Import wallet');
+
+    expect(invalidWord).toBeInvalid();
+    expect(importButton).toBeDisabled();
```

Comment:
For consistency with other test (I also found the `.not` to be easier to read)

```suggestion
    expect(importButton).not.toBeEnabled();
```